pipeline.go

```diff
@@ -128,9 +128,13 @@ func processResponse(pbResp *pipelinepb.QueryResults, tokens ...*pipelinepb.Toke
 		results = append(results, r)
 	}
 
-	d, err := ptypes.Duration(pbResp.GetLatency())
-	if err != nil {
-		return nil, err
+	var d time.Duration
+	var err error
```

Comment:
I think you could scope `var err error` to inside the block below so it can't be used by the rest of the method.

Reply:
Minor, but the code blocks below seem to do some checking and then assign directly onto `resp`. Not sure if it's worth following this same pattern for latency:

```go
{
	// ...
	if pbA := pbResp.GetAggregates(); pbA != nil {
		ra, err := processAggregatesResultMap(pbA)
		if err != nil {
			return nil, err
		}
		resp.Aggregates = ra
	}
	if pbA := pbResp.GetAggregateFilters(); pbA != nil {
		ra, err := processAggregatesResultMap(pbA)
		if err != nil {
			return nil, err
		}
		resp.AggregateFilters = ra
	}
}
```